Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,101 @@
+import AsyncHTTPClient
+import AWSLambdaRuntime
+import AWSLambdaEvents
+import DiscordBM
+import Foundation
+import SotoSecretsManager
+import GHHooksLambda
+import Models
+
+@main
+struct GHOAuthHandler: LambdaHandler {
+    typealias Event = APIGatewayV2Request
+    typealias Output = APIGatewayV2Response
+
+    let client: HTTPClient
+    let logger: Logger
+    let secretsRetriever: SecretsRetriever
+
+    init(context: LambdaInitializationContext) {
+        self.client = HTTPClient(eventLoopGroupProvider: .shared(context.eventLoop))
+        self.logger = context.logger
+        let awsClient = AWSClient(httpClientProvider: .shared(client))
+        self.secretsRetriever = SecretsRetriever(awsClient: awsClient, logger: logger)
+    }
+
+    func handle(
+        _ event: APIGatewayV2Request,
+        context: LambdaContext
+    ) async throws -> APIGatewayV2Response {
+        logger.trace("Received event: \(event)")
+
+        guard let code = event.queryStringParameters?["code"] else {
+            return .init(
+                statusCode: .badRequest,
+                body: "Missing code query parameter"
+            )
+        }
+
+        logger.trace("Retrieving secrets")
+        let clientID = try await secretsRetriever.getSecret(arnEnvVarKey: "BOT_TOKEN_ARN")
+        let clientSecret = try await secretsRetriever.getSecret(arnEnvVarKey: "WH_SECRET_ARN")
```

Comment:
Client ID and Client secret are totally different from these.

@gwynne will need to create the oauth app, then add the Client Secret to AWS secrets manager, and also give us the Client ID.

in the stack file it'll look something like this: https://github.com/vapor/penny-bot/blob/main/deploy/penny-discord-bot-stack.yml#L394

More precisely:
```yaml
Environment:
  Variables:
    CLIENT_SECRET_ARN: arn:aws:secretsmanager:eu-west-1:177420307256:secret:prod/penny/penny-bot/FIXME_ADD_ARN_WHEN_GWYNNE_ADDS_TO_SECRETS_MANANGER
    CLIENT_ID: # just plain id like: 213230123910312
``` 

Then:
```swift
let clientSecret = try await secretsRetriever.getSecret(arnEnvVarKey: "CLIENT_SECRET_ARN")
guard let clientID = ProcessInfo.processInfo.environment["CLIENT_ID"] else {
    throw Errors.envVarNotFound(name: "CLIENT_ID")
}
```